src/GraphQL.Tests/Extensions/TypeExtensionTests.cs

```diff
@@ -0,0 +1,100 @@
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+using GraphQL.Types.Relay.DataObjects;
+
+namespace GraphQL.Tests.Extensions;
+
+[Collection("StaticTests")]
+public class TypeExtensionTests
+{
+    [Theory]
+
+    // v8 naming
+    [InlineData(false, false, typeof(Class1), "Class1")]
+    [InlineData(false, false, typeof(Class1Type), "Class1")]
+    [InlineData(false, false, typeof(Class1Graph), "Class1Graph")]
+    [InlineData(false, false, typeof(Class1GraphType), "Class1")]
+    [InlineData(false, false, typeof(GenericClass<>), "TGenericClass")]
```

Comment:
Does it make sense to create a name for an open generic?

Reply:
I can't imagine a situation where `GetGraphQLName` is useful for an open generic type.  The test just demonstrates that if you were to do so, it will not crash and come up with the best name it can, based on the name of the type parameter, which is in this case "T".